[Docs] Adds language design to contributing section and restructures it a bit (#9773)
ethereum/solidity: docs/contributing.rst

Contributing
############

> Help is always appreciated and there are plenty of options how you can contribute to Solidity.

Reviewer: ```suggestion
Help is always appreciated and there are plenty of ways to contribute to Solidity.
```

> In particular, we appreciate support in the following areas:

Reviewer 2: Word repetition: appreciate

Author: Changed to `help is always welcome`. Hope that works. : )

* Reporting issues.
* Fixing and responding to `Solidity's GitHub issues
  <https://github.com/ethereum/solidity/issues>`_, especially those tagged as
  `good first issue <https://github.com/ethereum/solidity/labels/good%20first%20issue>`_ which are
  meant as introductory issues for external contributors.
* Improving the documentation.
* Translating the documentation into more languages.
* Responding to questions from other users on `StackExchange
  <https://ethereum.stackexchange.com>`_ and the `Solidity Gitter Chat
  <https://gitter.im/ethereum/solidity>`_.
* Getting involved in the language design process by joining language design calls, proposing language changes or new features and providing feedback.

> To get started, you can try :ref:`building-from-source` in order to familiarize

Reviewer: ```suggestion
If you wish to contribute code, you can start with :ref:`building-from-source` in order to familiarize
```
Something like that? Since the bullets above wouldn't necessarily require this step

Documentation Style Guide
=========================

> In the following section you find style recommendations specifically focusing on documentation

Reviewer 3: ```suggestion
In the following section you will find style recommendations specifically focusing on documentation
```

Make sure your contributions pass our documentation tests by running ``./scripts/docs.sh`` that installs dependencies
needed for documentation and checks for any problems such as broken links or syntax issues.

Solidity Language Design
========================

If you want to get involved in the language design process and share your ideas, please join the `solidity-users forum <https://groups.google.com/g/solidity-users>`_,

> where proposals for new language features and qualities of existing aspects of the language can be discussed.

Reviewer: `qualities of existing aspects` sounds a bit odd to me

Author: True, we could just say `where aspects of the language and proposals for new language features can be discussed`?

Author: or better `where existing properties of the language and proposals for new....`?

We regularly host language design discussion calls, in which selected topics, issues or feature implementations are debated in detail. The invitation
to those calls is shared via the aforementioned forum. We are also sharing feedback surveys and other language design relevant content in this forum.

For ad-hoc cases and questions you can reach out to us via the `Solidity-dev Gitter channel <https://gitter.im/ethereum/solidity-dev>`_, a
dedicated chatroom for conversations around the Solidity compiler and language development.

You can follow the implementation status of new features in the `Solidity Github project <https://github.com/ethereum/solidity/projects/43>`_.
Issues in the design backlog need further specification and will either be discussed in a language design call or in a regular team call. You can

> see the upcoming changes for the next breaking release by switching from the default branch (`develop`) to the `breaking branch <https://github.com/ethereum/solidity/tree/breaking>`_.

Reviewer 3: In RST you need double backticks to get the same formatting as in Markdown:

```suggestion
see the upcoming changes for the next breaking release by switching from the default branch (``develop``) to the `breaking branch <https://github.com/ethereum/solidity/tree/breaking>`_.
```
